lib/job.js

```diff
@@ -22,20 +26,15 @@ var Job = function(queue, name, data, opts){
     name = '__default__';
   }
 
-  opts = opts || {};
+  this.opts = _.extend({}, opts);
```

Comment:
I will update using _.defaults since it is also semantically better. I feel options should always be kept in the opts field only, so delay should be in opts instead of a property of job, however, delay can also be defined when retrying a job, and in that case it would not be an option but a property used by the queue mechanics. Timestamp is also a bit special since normally it should be an internal property, but  sometimes you want to override it, just as job.id.